package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -60,10 +84,112 @@ def to_json(self) -> str:
         return json.dumps(session_dict)
 
     def save(self):
-        """Save the session store info on db ."""
+        """Save the session store info on db and uploads it to the cloud if a remote cloud path is provided ."""
         engine, session_class = create_db_engine(self.location)
         Base.metadata.create_all(bind=engine)
         database = next(get_db(session_class))
-        session_store_data = RunModel(id=self._session_id, blob=self.to_json())
+
+        session_store_data = RunModel(id=self._session_id, blob=self._to_json())
         database.add(session_store_data)
         database.commit()
+        self.sync()
```

Comment:
> This is coming on well, great work! But I think there's two big things to consider still:
> 
> 1. what happens if my internet is broken? Everything should still work ok and not crash, but e.g. the `upload` method I think will break.
> 2. where is the logic to avoid duplicating rows in the database? i.e. the thing that says "only insert rows into my database if I don't already have them". Otherwise we'll just be repeatedly adding more and more rows to the database that we already have?

Not sure about point 1. it's a bit confusing. let me have a think and revert.

for your point 2. there's a unique constraint on run_ids in runs table so it won't add a run with the same timestamp. It throws an error which is why I have to do session.rollback()